mathesar_ui/src/component-library/render/Render.svelte

```diff
@@ -0,0 +1,17 @@
+<script lang="ts">
+  import type { SvelteComponent } from 'svelte';
+
+  import type { ComponentWithProps } from '../types';
+
+  type T = $$Generic<SvelteComponent>;
+
+  export let componentWithProps: ComponentWithProps<T>;
+
+  // I'm not sure how to fix the use of `any` here. Suggestions welcome!
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  $: component = componentWithProps.component as any;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  $: props = componentWithProps.props as any;
+</script>
+
+<svelte:component this={component} {...props} />
```

Comment:
@pavish, it sounds like you're proposing we get rid of this component and inline its definition into all of its call sites. I'm open to that — if we can figure out a clean way to do it. This component is only called in one place right now, but I imagine using it more. Basically any time we have a `ComponentWithProps` instance, this component renders it in a type-safe way.

With this component, we do:

```html
<Render componentWithProps={foo}/>
```

Without this component, we would do:

```html
<svelte:component this={foo.component} {...foo.props} />
```

That's a bit more verbose, but not terrible. The main problem is this type error I'm getting when I try to do that:

![image](https://github.com/mathesar-foundation/mathesar/assets/42411/59390bb5-9312-4c44-8423-927875f483f1)

I could be wrong, but this seems like a limitation of the typing provided by svelte-language-tools. It could be worth reporting this upstream to see what advice others have. I just didn't want to spend time on that right now.

Basically: wrapping the usage of `any` into a component seemed like a better approach to me because that way the dirty stuff is all wrapped up into one place that we can fix later on, even if we end up needing to use this in several places throughout the app.

Feel free to tinker with this and propose something better though.

Reply:
Also I added c30b99f513910b1afadc0c67249effe2661b6610 which I hope cleans this up a bit more.

Reply:
> wrapping the usage of any into a component seemed like a better approach to me because that way the dirty stuff is all wrapped up into one place that we can fix later on, even if we end up needing to use this in several places throughout the app

Okay, this seems good for now. `svelte-language-tools` doesn't seem to work great with generics and SvelteComponentTyped. I would like for us to figure out a better solution for this, probably, after a svelte version upgrade.